plugins/woocommerce/tests/e2e-pw/utils/themes.js

```diff
@@ -1,18 +1,42 @@
-const { exec } = require( 'node:child_process' );
+const { request } = require( '@playwright/test' );
+const { encodeCredentials } = require( '../utils/plugin-utils' );
+const { admin } = require( '../test-data/data' );
 
 export const DEFAULT_THEME = 'twentytwentythree';
 
-export const activateTheme = ( themeName ) => {
-	return new Promise( ( resolve, reject ) => {
-		const command = `wp-env run tests-cli wp theme install ${ themeName } --activate`;
+export const activateTheme = async ( baseURL, theme ) => {
+	const requestContext = await request.newContext( {
+		baseURL,
+		extraHTTPHeaders: {
+			Authorization: `Basic ${ encodeCredentials(
+				admin.username,
+				admin.password
+			) }`,
+			cookie: '',
+		},
+	} );
 
-		exec( command, ( error, stdout ) => {
-			if ( error ) {
-				console.error( `Error executing command: ${ error }` );
-				return reject( error );
+	try {
+		const response = await requestContext.post(
+			'/wp-json/e2e-theme/activate',
+			{
+				data: {
+					theme_name: theme,
+				},
 			}
+		);
 
-			resolve( stdout );
-		} );
-	} );
+		const result = await response.json();
+
+		if ( ! response.ok() ) {
+			throw new Error( `Failed to activate theme: ${ result.message }` );
+		}
+
+		return result;
+	} catch ( error ) {
+		console.error( 'Error activating theme:', error );
+		throw error;
+	} finally {
+		await requestContext.dispose(); // Clean up
+	}
```

Comment:
A very strange approach you got here, explicitly throwing the error in the try blocks to catch it an throwing it again. Why do we need the `requestContext.dispose()`?

How about simply throwing the error is response is not OK? This will cover the case where the theme does not exists or other API failures.

```
const result = await response.json();
if ( ! response.ok() ) {
	throw new Error( result.message );
}
return result;
```

Reply:
I forgot to revert it back as I added it only for testing purposes to see if the theme exists or not.

I can try that approach in a new commit.

Reply:
It's done